src/script/buildcontainer-setup.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+install_container_deps() {
+    source ./src/script/run-make.sh
+    prepare
+}
+
+dnf_clean() {
+    if [ "${CLEAN_DNF}" != no ]; then
+        dnf clean all
+        rm -rf /var/cache/dnf/*
+    fi
+}
+
+set -e
+export LOCALE=C
+cd ${CEPH_CTR_SRC}
+
+case "${CEPH_BRANCH}~${DISTRO}" in
+    *~*centos*stream8)
+        dnf install -y java-1.8.0-openjdk-headless /usr/bin/rpmbuild wget
+        install_container_deps
+        dnf_clean
+    ;;
+    *~*centos*stream9|*~*centos*stream10*)
+        dnf install -y /usr/bin/rpmbuild wget
+        install_container_deps
+        dnf_clean
+    ;;
+    *~*ubuntu*2?.04)
+        apt-get update
```

Comment:
is there any reason to run update here? and not use the packages in the base image?

Reply:
`apt-get update` is needed to download the repo metadata prior to running `apt-get install`. at least that's what I think was the case... but my debian is a bit rusty.

Reply:
my concern on this is 2 fold
1. we will be (afaik) updating all the packages to the latest available one (do we want this to happen every time? i would think we rely on the base ubuntu image being updated?)
2. we dont do this for others - regardless of what we pick - we should do it consistently across the various OS/images (ie run `update` OR just use the latest base image)

Reply:
> my concern on this is 2 fold
> 
>     1. we will be (afaik) updating all the packages to the latest available one (do we want this to happen every time? i would think we rely on the base ubuntu image being updated?)
> 

I am willing to be corrected (as always) if I am wrong but, and I did just check this locally, calling `apt-get update` doesn't update packages (`apt-get upgrade` does), this *only* downloads metadata.
This does cause it to get the latest metadata, and not a snapshot of the metadata at the point the container was originally created, but I don't think that's what you are concerned about right?

As an example, here's a brief example I just ran to assure myself of what I just wrote:
```
[john@edfu ~]$ podman run -it --rm docker.io/library/ubuntu:24.04
root@f358f91cace7:/# apt-get install cowsay
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package cowsay
root@f358f91cace7:/# 
exit

[john@edfu ~]$ podman run -it --rm docker.io/library/ubuntu:24.04
root@7bc0a8502e6c:/# apt-get update
Get:1 http://security.ubuntu.com/ubuntu noble-security InRelease [126 kB]
Get:2 http://archive.ubuntu.com/ubuntu noble InRelease [256 kB]
Get:3 http://archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Get:4 http://archive.ubuntu.com/ubuntu noble-backports InRelease [126 kB]
Get:5 http://security.ubuntu.com/ubuntu noble-security/main amd64 Packages [630 kB]
Get:6 http://security.ubuntu.com/ubuntu noble-security/multiverse amd64 Packages [15.3 kB]
Get:7 http://security.ubuntu.com/ubuntu noble-security/restricted amd64 Packages [609 kB]
Get:8 http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages [19.3 MB]
Get:9 http://security.ubuntu.com/ubuntu noble-security/universe amd64 Packages [728 kB]
Get:10 http://archive.ubuntu.com/ubuntu noble/main amd64 Packages [1808 kB]            
Get:11 http://archive.ubuntu.com/ubuntu noble/restricted amd64 Packages [117 kB]
Get:12 http://archive.ubuntu.com/ubuntu noble/multiverse amd64 Packages [331 kB]
Get:13 http://archive.ubuntu.com/ubuntu noble-updates/universe amd64 Packages [944 kB]
Get:14 http://archive.ubuntu.com/ubuntu noble-updates/main amd64 Packages [891 kB]
Get:15 http://archive.ubuntu.com/ubuntu noble-updates/multiverse amd64 Packages [19.7 kB]
Get:16 http://archive.ubuntu.com/ubuntu noble-updates/restricted amd64 Packages [679 kB]
Get:17 http://archive.ubuntu.com/ubuntu noble-backports/universe amd64 Packages [11.9 kB]
Fetched 26.7 MB in 2s (12.1 MB/s)                          
Reading package lists... Done
root@7bc0a8502e6c:/# apt-get install cowsay
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libgdbm-compat4t64 libgdbm6t64 libperl5.38t64 libtext-charwidth-perl netbase perl
  perl-modules-5.38
Suggested packages:
  filters cowsay-off gdbm-l10n perl-doc libterm-readline-gnu-perl
  | libterm-readline-perl-perl make libtap-harness-archive-perl
The following NEW packages will be installed:
  cowsay libgdbm-compat4t64 libgdbm6t64 libperl5.38t64 libtext-charwidth-perl netbase
  perl perl-modules-5.38
0 upgraded, 8 newly installed, 0 to remove and 6 not upgraded.
Need to get 8297 kB of archives.
After this operation, 50.6 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
```

Note that the debian-style package commands do not automatically download the package metadata, unlike `dnf` from the rpm ecosystem. Thus calling `apt-get update` at least once before installing a package is needed. However, it doesn't change the installed packages by itself.

>     2. we dont do this for others - regardless of what we pick - we should do it consistently across the various OS/images (ie run `update` OR just use the latest base image)

I'm not sure what "others" you are referring to here are. Could you please elaborate?

Reply:
by other - i meant `dnf update`

if this is just doing metadata update for the following installs (quirks for `apt-get`) - this makes sense!

(things i learned today! ❤️ )